src/hyperion/external_interaction/callbacks/xray_centre/ispyb_callback.py

```diff
@@ -97,35 +105,38 @@ def activity_gated_start(self, doc: RunStart):
                 self.params.hyperion_params.detector_params,
                 self.params.hyperion_params.ispyb_params,
             )
-            grid_scan_info = GridScanInfo(
-                self.params.hyperion_params.ispyb_params.upper_left,
-                self.params.experiment_params.y_steps,
-                self.params.experiment_params.y_step_size,
-            )
 
-            def constructor():
-                return construct_comment_for_gridscan(
-                    self.params,
-                    self.params.hyperion_params.ispyb_params,
-                    grid_scan_info,
-                )
-
-            scan_data_info = ScanDataInfo(
-                data_collection_info=populate_remaining_data_collection_info(
-                    constructor,
-                    None,
-                    populate_xy_data_collection_info(
-                        grid_scan_info,
-                        self.params,
-                        self.params.hyperion_params.ispyb_params,
+            scan_data_infos = [
```

Comment:
Should: Rather than do anything specific upfront could we get away with doing all the specific stuff in `_handle_oav_snapshot_triggered`? e.g. this could be something like:
```python
number_of_collections = 2 if self.is_3d_gridscan() else 1

scan_data_infos = [ScanDataInfo(data_collection_info=populate_remaining_data_collection_info(
                            None,
                            None,
                            DataCollectionInfo(
                                axis_range=0,
                                axis_end=omega_start,
                            ),
                            self.params.hyperion_params.detector_params,
                            self.params.hyperion_params.ispyb_params,
                        ) for range in (number_of_collections)]
```

Then handle `run_number` and `omega_start` in `_handle_oav_snapshot_triggered`?

Reply:
I guess this is a wider question than for this one particular change but given that we are planning to move to ispyb-eye is it better to defer some of our internal changes until that point? At least until it's clearer how start / event documents might map to ispyb-eye calls? Unless you were proposing this with that already in mind.

Reply:
Sure, happy to park this :)